bibigrid-core/src/main/java/de/unibi/cebitec/bibigrid/core/intents/IdeIntent.java

```diff
@@ -86,9 +91,24 @@ private void startPortForwarding(final String masterIp) {
         }
     }
 
+    /**
+     * Checks if IDE port (default 8181) is already binded.
+     * Uses next port (+1) if available
+     * @param port check if port is already listened on
+     */
+    private void setIDEPort(int port) {
+        try (ServerSocket serverSocket = new ServerSocket()) {
+            // serverSocket.setReuseAddress(false); required only on OSX
+            serverSocket.bind(new InetSocketAddress(InetAddress.getByName("localhost"), port), 1);
+            this.idePort = port;
+        } catch (Exception ex) {
+            setIDEPort(++port);
```

Comment:
What happens if all ports are occupied ?  The recursive call of setIDEPort is nice, but resource intensive if a lot of ports have to be checked ...
There should be a upper limit of how much ports should be checked before abort.
Maybe it is option to leave this to the user (configration properties : port start, port end)